go/dhcp/main.go

```diff
@@ -308,7 +307,13 @@ func (h *Interface) ServeDHCP(ctx context.Context, p dhcp.Packet, msgType dhcp.M
 			firstTry := true
 			log.LoggerWContext(ctx).Info("DHCPDISCOVER from " + clientMac + " (" + clientHostname + ")")
 			var free int
-
+			// Static assign IP address ?
+			for macaddr, position := range handler.ipAssigned {
```

Comment:
This would be way faster if ipAssigned was a map and you performed a lookup based on the MAC instead of an iteration.

Otherwise, the more you add reserved IPs, the slower this gets